### ISSUE TITLE
"Connection reset by peer" after 32768 bytes

### ISSUE THREAD

**Issue by tYYGH** [solusipse/fiche#24]
Example:
```
~$ ncat localhost 9999 <.pine-debug1
http://localhost/f/8gau5w
Ncat: Connection reset by peer.
~$ ls -l /var/www/fiche/8gau5w/index.txt
-rw-r--r-- 1 www-data www-data 32768 nov.   4 11:57 /var/www/fiche/8gau5w/index.txt
~$ ls -l .pine-debug1
-rw------- 1 tYYGH users 115080 oct.   7 10:46 .pine-debug1
```
This is on Debian-stable (ARMv5 32 bits). Fiche is launched like this:
```
fiche -d localhost/f -o /var/www/fiche -p 9999 -s 6 -u www-data -l /var/log/fiche.log
```
The log file shows no errors.

**Comment by solusipse** [comment]
It's not a bug, it's a feature! But seriously, it was designed to work like that. Use `-B` option to increase maximum file size. By default it's 32768. For example, on termbin.com it's 1MB.

**Comment by tYYGH** [comment]
@solusipse, Thank you for the explanation.
I see in fiche’s help that -B gives the “buffer size”. Does it mean that if I increase the -B value in order to allow bigger documents, fiche would then allocate a bigger chunk of memory in a permanent way? For example, if I were to set this option to “-B 10000000” (10MB), would it imply that fiche would allocate a permament 10MB of RAM, thus making my server’s memory 10MB less?

**Comment by solusipse** [comment]
Yep. If that's a problem, this can be changed easily, just decrease buffer size and put `recv` inside a loop (and put page generation logic there). If I remember correctly, some old netcat implementations may not work correctly anymore. I never really cared about memory consumption of this program, it's still minimal for modern servers. See my other project, [ureq](https://github.com/solusipse/ureq), where memory usage was the biggest concern, I implemented there a solution I described here. Thanks for the question.
